core/src/test/java/org/hibernate/ogm/backendtck/storedprocedures/indexed/IndexedStoredProcedure.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.storedprocedures.indexed;
+
+import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
+import org.hibernate.ogm.model.spi.Tuple;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public interface IndexedStoredProcedure {
+	ClosableIterator<Tuple> execute(Object[] params);
```

Comment:
I understand that this is only used for test but couldn't you use `Function<Object[], ClosableIterator<Tuple>>` instead? I'm asking because I would prefer not to have unnecessary interfaces

Reply:
Do you mean using java.util.function.Function ?

Reply:
Yes, but you know what, it's not critical. Let's leave this

Reply:
Ok. just see your last comment. Sorry.

Reply:
[1:40 PM] Davide D'Alto: Yes, we switched to Java 8, so it's OK to use the new classes every once in a while
[1:44 PM] Davide D'Alto: The interface you added is actually very generic, I think it would be ok to use a Funciton class in that specific scenario

Reply:
fixed.